### DISCUSSION TITLE
opset_version uncompatible -- fail to load ort file on Android,

### DISCUSSION THREAD

**Discussion by theta666** [snakers4/silero-vad#103] [Q&A]
I just converted the onnx file to ort file according to onnxruntime official webset, and when loading it(create session) in code with pre-build runtime on Android, AndroidStudio reports this error message: Failed to find kernel for Pow(7) (node Pow_874), I checked the onnxruntime-mobile type sheet [support](https://onnxruntime.ai/docs/reference/operators/mobile_package_op_type_support_1.9.html), of which the Pow opset is {12,13}, SO is there any way to re-create onnx file with newer opset_version?

**Comment by snakers4** [comment]
@theta666 
Which model did you try?
Can you maybe try this - https://github.com/snakers4/silero-vad/issues/37#issuecomment-913629296 ?

@adamnsandle 
Which opset did you use during export, 11?

**Reply by snakers4** [reply]
We can export the next models with newer opsets

**Reply by theta666** [reply]
I tried this [model.onnx](https://github.com/snakers4/silero-vad/blob/master/files/model.onnx),  during the conversion strage, a config file generated like this
`ai.onnx;1;GlobalAveragePool,Identity,LayerNormalization,Shape,Transpose

ai.onnx;5;Reshape

ai.onnx;6;Log,Neg,Relu,Sigmoid,Sqrt

ai.onnx;7;Add,Div,Mul,Pow

ai.onnx;9;MatMul

ai.onnx;11;Concat,Conv,Equal,Gather,If,Pad,ReduceMean,Slice,Softmax,Split,Squeeze,Unsqueeze`
it seems that there are various opset versions in the ort file, I'm new to onnx, not sure whether it's because the uncompatible version thing, 
thanks for reply, I will try the jit  if no simple solution for onnxruntime-mobile

**Reply by snakers4** [reply]
You are the first person reporting to use the mobile ONNX builds (though someone launched on ARM) and I believe they are quite new anyway

**Reply by snakers4** [reply]
Can you try with new ONNX model?

**Reply by theta666** [reply]
I tried the onnx model inside 3.1,  the new onnx model fails to run on android due to similar reason:
detail:
```
# convert command
python3 -m onnxruntime.tools.convert_onnx_models_to_ort --optimization_level=all ./silero_vad.onnx
```
operation list during conversion:
```
ai.onnx;1;GlobalAveragePool,LayerNormalization
ai.onnx;7;LSTM
ai.onnx;11;Conv
ai.onnx;13;Add,Concat,Div,Equal,Gather,Identity,If,Log,MatMul,Mul,Neg,Pad,Pow,ReduceMean,Relu,Reshape,Shape,Sigmoid,Slice,Softmax,Sqrt,Squeeze,Transpose,Unsqueeze
com.microsoft;1;FusedConv,FusedMatMul
```
when loading the converted ort file on android, it shows this error:
```
ai.onnxruntime.OrtException: Error code - ORT_FAIL - message: inference_session.cc:1217 AssignNodesToEpsFromHashesImpl Failed to find kernel def hash (5860801277476873352) in kernel registries for LSTM(7) node with name 'LSTM_600'.
```
it seems that the "com.microsoft.onnxruntime:onnxruntime-mobile:1.9.0" does not support lstm operation in onnx model

**Reply by snakers4** [reply]
is it supported in some newer opsets maybe?

**Reply by theta666** [reply]
no, according to the official doc, [https://onnxruntime.ai/docs/reference/operators/mobile_package_op_type_support_1.10.html](https://onnxruntime.ai/docs/reference/operators/mobile_package_op_type_support_1.10.html)
maybe there's a way of convert lstm to custom op, I'm not professional on this

**Reply by snakers4** [reply]
looks like they do not support LSTM in mobile yet
basically LSTM is just a combination of linear layers and gates, but I do not believe we would have time and resources to implement custom LSTM just for the sake of being exportable to ONNX mobile
